aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1604,6 +1606,125 @@ Tensor& linalg_norm_out(Tensor& result, const Tensor& self, std::string ord, opt
   return linalg_norm_out_impl(result, self, c10::nullopt, ord, opt_dim, keepdim, opt_dtype);
 }
 
+Tensor _linalg_cond_exception_helper(const Tensor& self) {
+  // For batched input if at least one matrix in the batch is not invertible,
+  // we can't get the result for all other (possibly) invertible matrices in the batch without an explicit for loop.
+  // This should change when at::inverse works with silent errors
+  if (self.dim() > 2) {
+    TORCH_CHECK(false,
+      "One or more matrices in the batch was not invertible! "
+      "linalg_cond does not support yet this case.");
+  }
+  auto result_shape = self.sizes().vec();
+  result_shape.pop_back();
+  result_shape.pop_back();  // result's shape is equal to self.shape[0:-2]
+  Tensor result = at::full(result_shape, INFINITY, self.options());
+  return result;
+}
+
+// This function helps to dispatch norm computations depending on 'ord' of variant type
+Tensor _linalg_cond_helper(const Tensor& self, c10::variant<Scalar, std::string> ord_variant) {
+  // Ignore errors if not invertible, result is INFINITY in this case
+  // Currently checking for error in at::inverse causes cross-device data movement
+  // For batched input if at least one matrix in the batch is not invertible,
+  // then the result for all other (possibly) invertible matrices will be infinity as well
+  // since there is currently no way to use at::inverse with silent errors
+  Tensor self_inverse;
+  try {
+    self_inverse = at::inverse(self);
+  } catch (const std::exception& e) {
+    if (strstr(e.what(), "singular")) {
+      return _linalg_cond_exception_helper(self);
+    } else {
+      TORCH_CHECK(false, "linalg_cond got an unexpected error:\n", e.what());
+    }
+  }
+  std::array<int64_t, 2> dim_arr = {-2, -1};
+  optional<IntArrayRef> dim = IntArrayRef(dim_arr);
+
+  return c10::visit([&](auto&& ord) {
+    Tensor norm_self = at::linalg_norm(self, ord, dim);
+    Tensor norm_inverse = at::linalg_norm(self_inverse, ord, dim);
+    Tensor result = norm_self * norm_inverse;
+    return result;
+  }, ord_variant);
+}
+
+// Numerical or None norms
+Tensor linalg_cond(const Tensor& self, optional<Scalar> opt_ord) {
+  TORCH_CHECK(self.numel() > 0, "linalg_cond is not defined for empty tensors.");
```

Comment:
It makes sense to factor out shape checks for both overloads. Also, numpy accepts empty batch size, so `numel==0` is a wrong check:
```
In [11]: a=np.arange(0).reshape(0,3,3).astype(np.float)                                                                                                                                                             

In [12]: a                                                                                                                                                                                                          
Out[12]: array([], shape=(0, 3, 3), dtype=float64)

In [13]: LA.cond(a)                                                                                                                                                                                                 
Out[13]: array([], dtype=float64)
```

Reply:
Okay, I didn't check the NumPy's behavior for empty batch size, thank you!
NumPy doesn't work for empty matrices https://github.com/numpy/numpy/blob/master/numpy/linalg/linalg.py#L1762-L1763.

Reply:
It's likely that someone will ask we support empty matrices, too, and this PR can add that support as well (and test it and note in the tests its divergent from NumPy).

Reply:
I added support for 0x0 matrices in https://github.com/pytorch/pytorch/pull/45832/commits/b15fedda0078a4a544f3f2f44e75a46288dc0eba.